app/tests/components_tests/test_views.py

```diff
@@ -539,3 +543,74 @@ def test_interfaces_list_link_in_new_interface_form(
         user=u,
     )
     assert reverse(interface_list_viewname) in response.rendered_content
+
+
+@pytest.mark.parametrize(
+    "base_object_factory,base_obj_lookup,object_factory,viewname",
+    (
+        (
+            ReaderStudyFactory,
+            "reader_study",
+            DisplaySetFactory,
+            "reader-studies:display-set-update",
+        ),
+        (
+            ArchiveFactory,
+            "archive",
+            ArchiveItemFactory,
+            "archives:item-edit",
+        ),
+    ),
+)
+@pytest.mark.django_db
+def test_image_widget_populated_value_on_update_view_validation_error(
+    client, base_object_factory, base_obj_lookup, object_factory, viewname
+):
+    image1 = ImageFactory()
+    image_ci = ComponentInterfaceFactory(kind=ComponentInterface.Kind.IMAGE)
+    image1_civ = ComponentInterfaceValueFactory(
+        interface=image_ci, image=image1
+    )
+
+    annotation = "{}"
+    annotation_ci = ComponentInterfaceFactory(
+        kind=ComponentInterface.Kind.TWO_D_BOUNDING_BOX
+    )
+    annotation_civ = ComponentInterfaceValueFactory(
+        interface=annotation_ci, value=annotation
+    )
+
+    editor = UserFactory()
+    base_obj = base_object_factory()
+    base_obj.add_editor(editor)
+
+    object = object_factory(**{base_obj_lookup: base_obj})
+    object.values.set([image1_civ, annotation_civ])
+
+    image2 = ImageFactory()
+    data = {
+        "interface_slug": image_ci.slug,
+        "current_value": image2.pk,
+    }
+    data.update(
+        **get_interface_form_data(interface_slug=image_ci.slug, data=image2.pk)
+    )
+    data.update(
+        **get_interface_form_data(
+            interface_slug=annotation_ci.slug, data='{"1":1}'
+        )
+    )
```

Comment:
Why update the dictionary here and not just construct it in one go?

Reply:
the function `get_interface_form_data` already returns a dictionary and it is called twice for each field, so I used update() to append the two dictionaries.

Reply:
You're unpacking that dictionary. So you can just do 
```suggestion
    data = {
        "interface_slug": image_ci.slug,
        "current_value": image2.pk,
        **get_interface_form_data(interface_slug=image_ci.slug, data=image2.pk)
        **get_interface_form_data( interface_slug=annotation_ci.slug, data='{"1":1}')
    }
```

Reply:
much cleaner, thanks!